packages/plugin-typescript/sources/index.ts

```diff
@@ -25,6 +26,12 @@ const afterWorkspaceDependencyAddition = async (
   const {project} = workspace;
   const {configuration} = project;
 
+  const tsEnableAutoTypes = configuration.get(`tsEnableAutoTypes`)
+    ?? xfs.existsSync(ppath.join(project.cwd, `tsconfig.json` as Filename));
+
+  if (!tsEnableAutoTypes)
+    return;
```

Comment:
Should check this when removing dependencies as well

Reply:
I'm not entirely sure I agree with this - while adding `@types` should only happen when the user uses TS, they can be safely removed even when the user doesn't, since `@types/foo` will always be related to `foo`.